assets/dev/js/editor/controls/base.js

```diff
@@ -146,6 +146,35 @@ ControlBaseView = Marionette.CompositeView.extend( {
 
 		this.toggleControlVisibility();
 	},
+
+	reRoute( controlActive ) {
+		$e.route(
+			$e.routes.getCurrent( 'panel' ),
+			this.getControlInRouteArgs( controlActive ? this.getControlPath() : '' ),
+			{ history: false },
+		);
+	},
+
+	getControlInRouteArgs( path ) {
+		return {
+			...$e.routes.getCurrentArgs( 'panel' ),
+			activeControl: path,
+		};
+	},
+
+	getControlPath() {
+		let controlPath = this.model.get( 'name' ),
+			parent = this._parent;
+
+		while ( ! parent.$el.hasClass( 'elementor-controls-stack' ) ) {
+			const parentName = parent.model.get( 'name' ) || parent.model.get( '_id' );
+			controlPath = parentName + '/' + controlPath;
+
+			parent = parent._parent;
+		}
+
+		return controlPath;
+	},
```

Comment:
How does this code relate to the control? and if it is, why on the base control? Does each control have a route?

Reply:
The idea is here was to add to the routing system an "activeControl" argument which is a control path to the the active control in the current route. When a route is applied and it has an activeControl, then based on the controlPath of that control we first get its Control View and then call the activate method on it.
Each control does not have its own names route if that's what you meant, but each control has its own path and thus its own "activeControl" argument in a route.

Reply:
Talked about it with @Nevoss, waiting for resolve.